indico/modules/events/papers/controllers/management.py

```diff
@@ -141,3 +142,49 @@ def _process_GET(self):
     def _process_POST(self):
         self.list_generator.store_configuration()
         return jsonify_data(flash=False, **self.list_generator.render_list())
+
+
+class RHManagePapersActionsBase(RHManagePapersBase):
+    """Base class for RHs performing actions on selected paper contributions"""
+
+    _contrib_query_options = ()
+
+    @property
+    def _contrib_query(self):
+        query = Contribution.query.with_parent(self.event_new)
+        if self._contrib_query_options:
+            query = query.options(*self._contrib_query_options)
+        return query
+
+    def _checkParams(self, params):
+        RHManagePapersBase._checkParams(self, params)
+        self.list_generator = PaperAssignmentListGenerator(event=self.event_new)
+        ids = map(int, request.form.getlist('contribution_id'))
+        self.contributions = self._contrib_query.filter(Contribution.id.in_(ids)).all()
+
+
+class RHBulkPaperStateChange(RHManagePapersActionsBase):
+    """Perform bulk change state operations on selected papers"""
+
+    def _process(self):
+        form = BulkPaperJudgmentForm(event=self.event_new, judgment=request.form.get('judgment'),
+                                     contribution_id=[a.id for a in self.contributions])
```

Comment:
Copied from abstracts I guess? ;)

`a` doesn't fit here much, I'd use `c` or the generic `x` (but usually I go for `c` for contributions)